File: clients/ruby/model/specialty.rb
module Model

# Player's specialty
module Specialty
    # Logistics. Increased travel distance
    LOGISTICS = 0
    # Production. Increased work speed
    PRODUCTION = 1
    # Combat. Increased damage
    COMBAT = 2

    # Read Specialty from input stream
    def self.read_from(stream)
        result = stream.read_int()
        if result < 0 || result >= 3
            raise "Unexpected tag value"
        end
        result
    end

    def self.to_s(value)
        if value == LOGISTICS
            return "LOGISTICS"
        end
        if value == PRODUCTION
            return "PRODUCTION"
        end
        if value == COMBAT
            return "COMBAT"
        end
        raise "Impossible happened"
    end

    def self.to_str(value)
        self.to_s(value)
    end
end

end

File: clients/go/model/specialty.go
package model

import "io"
import . "spb_ai_champ/stream"

// Player's specialty
type Specialty int32

const (
    // Logistics. Increased travel distance
    SpecialtyLogistics Specialty = 0
    // Production. Increased work speed
    SpecialtyProduction Specialty = 1
    // Combat. Increased damage
    SpecialtyCombat Specialty = 2
)

// Read Specialty from reader
func ReadSpecialty(reader io.Reader) Specialty {
    switch ReadInt32(reader) {
    case 0:
        return SpecialtyLogistics
    case 1:
        return SpecialtyProduction
    case 2:
        return SpecialtyCombat
    }
    panic("Unexpected tag value")
}

// Get string representation of Specialty
func SpecialtyToString(specialty Specialty) string {
    switch specialty {
    case SpecialtyLogistics:
        return "Logistics"
    case SpecialtyProduction:
        return "Production"
    case SpecialtyCombat:
        return "Combat"
    }
    panic("Impossible happened")
}